### PR TITLE
Save hosts in listener

### PR DESCRIPTION
**Pull request by semtexzv**
Moves code which processes uploads to separate file.  Implements process to get system_profile from inventory, create VMaaS request, And store both RHAccount and SystemProfile into the database.

Tests for parts which do not require external APIs are implemented.

### CODE REVIEW

File: base/models/models.go [hunk 1]
```diff
@@ -69,6 +70,15 @@ type AdvisoryAccountData struct {
 func (AdvisoryAccountData) TableName() string {
 	return "advisory_account_data"
 }
+type SystemAdvisoriesSlice []SystemAdvisories
+
+func (this SystemAdvisoriesSlice) ToInterfaceSlice() []interface{} {
+	var res []interface{}
```
Comment (Josca) [inline base/models/models.go]: As we know source array size, could we allocate result array in the beginning?

Comment (Josca) [reply base/models/models.go]: The same optimization could be done in the `OnConflictUpdateMulti` method.

File: listener/upload.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	hash := sha256.New()
+	hash.Write(updatesReqJSON)
+	jsonChecksum := hex.EncodeToString(hash.Sum([]byte{}))
+
+	now := time.Now()
+
+	dbHost := models.SystemPlatform{
+		InventoryID:    inventoryId,
+		RhAccountID:    accountId,
+		FirstReported:  &now,
```
Comment (Josca) [inline listener/upload.go]: Is it sure, that this is the first upload?

File: listener/upload.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	"encoding/json"
+	"github.com/RedHatInsights/patchman-clients/inventory"
+	"github.com/RedHatInsights/patchman-clients/vmaas"
+	"github.com/antihax/optional"
+	"github.com/segmentio/kafka-go"
+	"time"
+)
+
+func uploadHandler(m kafka.Message) {
+	var event PlatformEvent
```
Comment (Josca) [inline listener/upload.go]: How can I test it? Is there any script or guide to run all workflow using docker-compose?

File: base/models/models.go [hunk 4]
```diff
@@ -51,7 +52,7 @@ type SystemAdvisories struct {
 	ID            int
 	SystemID      int
 	AdvisoryID    int
-	FirstReported time.Time
+	FirstReported *time.Time
```
Comment (Josca) [inline base/models/models.go]: This field corresponds to `first_reported` column in db which is NOT NULL, so it doesn't have to be pointer.

File: listener/listener.go [hunk 5]
```diff
@@ -46,19 +44,21 @@ func configure() {
...
 	eventsReader = kafka.NewReader(eventsConfig)
 
+	traceApi := utils.GetenvOrFail("LOG_LEVEL") == "trace"
+
 	inventoryAddr := utils.GetenvOrFail("INVENTORY_ADDRESS")
 
-	config := inventory.NewConfiguration()
-	config.Debug = true
-	config.BasePath = inventoryAddr + INVENTORY_API_PREFIX
+	invConfig := inventory.NewConfiguration()
```
Comment (Josca) [inline listener/listener.go]: Yes, these names are better. But I wouldn't use shortcut here, but name it simply `inventoryConfig`.

File: base/models/models.go (hunk 4 above)
Comment (semtexzv) [reply base/models/models.go]: If it isn't null its set to `0000-00- .....` By gorm. This is wrong. The field is automatically set to current timestamp by a database trigger if null is provided.

File: base/models/models.go (hunk 1 above)
Comment (semtexzv) [reply base/models/models.go]: Fixed, thanks

File: listener/upload.go (hunk 2 above)
Comment (semtexzv) [reply listener/upload.go]: This field is set by database trigger. Fixed.

File: listener/listener.go (hunk 5 above)
Comment (semtexzv) [reply listener/listener.go]: Fixed.

File: listener/upload.go (hunk 3 above)
Comment (semtexzv) [reply listener/upload.go]: Not yet, we'll need to either extend Unit tests to run with platform, or introduce separate tools for testing whole workflows ...

**Pull request merged**